app/prism/PrismLogic.scala

```diff
@@ -99,4 +101,26 @@ object PrismLogic {
       }
     }.getOrElse(true)
   }
+
+  /**
+    * @return Percentiles of instance AMIs age
+    */
+  def instancesAmisAgePercentiles(amis: List[AMI]): Percentiles = {
+    val ages = amis.flatMap { ami =>
+      val amiAge = ami.creationDate.map(DateUtils.daysAgo).getOrElse(0)
+      List.fill(ami.instances.getOrElse(Nil).length)(amiAge)
+    }
+    Percentiles(ages)
+  }
+
+  /**
+    * @return the number of instances from the list of AMIs that match a given SSA
+    */
+  def instancesCountForSSA(amis: List[AMI], ssa: SSA): Int = {
```

Comment:
Not used yet, but would be in a future PR.
You are right about the "hydrated" bit though. Any idea about how to improve the situation?